go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kind
 
-go 1.16
+go 1.21
```

Comment:
we've intentionally held this back, people `go get` / `go install` kind and use systems with outdated go and we have no particular need for latest go features in the core tool

Reply:
we've been hacking on bumping it to 1.17 for related reasons but not as far as 1.21 currently https://github.com/kubernetes-sigs/kind/pull/3335

Reply:
I've been waiting for @aojea to take another review pass at https://github.com/kubernetes-sigs/kind/pull/3335, but if we wanted to land a more targeted change, just bumping to 1.17 here will accomplish the same thing with less impact.

Reply:
They mustn’t use Go 1.20, as it has reached EOL and no longer receives security fixes.

Reply:
That's their choice, the kind binary is not particularly sensitive to these as a local non-networked CLI. We document that 1.16+ is required to install currently and have no use case for newer go features for the core project.

Being able to `go install` on e.g., debian stable, without issues, is something we do not want to break lightly.

Reply:
And again, to get the module pruning, you only need to increase this to 1.17

Reply:
We don't break users without a good reason, lots of new k8s users are installing the kind CLI via their system go and it works fine.

similarly we still have limited support for kubernetes versions that kubernetes no longer supports (e.g. so lagging users can test before/after updating), it would be easier to force only current versions but it has value to users not to and the burden is not too high

Reply:
Sure, closing this for niw